parentheses/parentheses.go

```diff
@@ -0,0 +1,67 @@
+package parentheses
+
+import (
+	"fmt"
+	"math/rand"
+	"net/http"
+	"strconv"
+)
+
+func IsBalanced(s string) bool {
+	stack := []rune{}
+	mapping := map[rune]rune{'(': ')', '[': ']', '{': '}'}
+
+	for _, c := range s {
+		if c == '(' || c == '[' || c == '{' {
+			stack = append(stack, c)
+		} else if c == ')' || c == ']' || c == '}' {
+			if len(stack) == 0 {
+				return false
+			}
+
+			last := stack[len(stack)-1]
+			stack = stack[:len(stack)-1]
+
+			if mapping[last] != c {
+				return false
+			}
+		}
+	}
+
+	return len(stack) == 0
+}
+
+func GenerateSequence(length int) string {
+	var parentheses string
+	for i := 0; i < length; i++ {
+		switch rand.Perm(6)[0] {
+		case 0:
+			parentheses += "("
+		case 1:
+			parentheses += ")"
+		case 2:
+			parentheses += "["
+		case 3:
+			parentheses += "]"
+		case 4:
+			parentheses += "{"
+		case 5:
+			parentheses += "}"
+		}
+	}
+	return parentheses
+}
+
+func GenerateSequenceHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
Good job separating these, now you should move the handler out of `parenthesis` package. 
It's better to have smaller packages that have a distinct purpose. `parenthesis` does some helpful operations with parenthesis. It shouldn't also provide an `http.Handler` for each of it's functions. 
There are a ton of ways to use `parenthesis.GenerateSequence()` you don't include usage into the package itself. Because otherwise you would need to also include a handler for protobuf, and a CLI handler, and a ton of other functions.

Normally you would have a separate package for handlers (for example `package "api"`). (It would also be ok to just move to `main`, since you only have 1 handler, but maybe having tests in main is not a great idea)